akka-docs-dev/rst/scala/stream-composition.rst

```diff
@@ -0,0 +1,223 @@
+.. _composition-scala:
+
+Modularity, Composition and Hierarchy
+=====================================
+
+Akka Streams provide a uniform model of stream processing graphs, which allows flexible composition of reusable
+components. In this chapter we show how these look like from the conceptual and API perspective, demonstrating
+the modularity aspects of the library.
+
+Basics of composition and modularity
+------------------------------------
+
+Every processing stage used in Akka Streams can be imagined as a "box" with input and output ports where elements to
+be processed arrive and leave the stage. In this view, a :class:`Source` is nothing else than a "box" with a single
+output port, or, a :class:`BidiFlow` is a "box" with exactly two input and two output ports. In the figure below
+we illustrate the most common used stages viewed as "boxes".
+
+|
+
+.. image:: ../images/compose_shapes.png
+   :align: center
+
+|
+
+The *linear* stages are :class:`Source`, :class:`Sink`
+and :class:`Flow`, as these can be used to compose strict chains of processing stages.
+Fan-in and Fan-out stages have usually multiple input or multiple output ports, therefore they allow to build
+more complex graph layouts, not just chains. :class:`BidiFlow` stages are usually useful in IO related tasks, where
+there are input and output channels to be handled. Due to the specific shape of :class:`BidiFlow` it is easy to
+stack them on top of each other to build a layered protocol for example. The ``TLS`` support in Akka is for example
+implemented as a :class:`BidiFlow`.
+
+It is easy to see that these reusable components already allow the creation of complex processing networks. What we
```

Comment:
I agree it's easy, but such wording may make someone feel stupid if they read all this stuff for the first time and are a bit overwhelmed. 

How about just `these reusable components already`